src/irust/cursor.rs

```diff
@@ -3,121 +3,136 @@ use crate::utils::StringTools;
 mod bounds;
 use bounds::Bounds;
 
+use crossterm::TerminalCursor;
+
 #[derive(PartialEq)]
 pub enum Move {
     Free,
     Modify,
 }
 
 #[derive(Clone)]
-pub struct Cursor {
+pub struct CursorPosition {
     pub screen_pos: (usize, usize),
     pub buffer_pos: usize,
     pub bounds: Bounds,
     pub lock_pos: (usize, usize),
+}
+
+pub struct Cursor {
+    pub pos: CursorPosition,
+    internal_cursor: TerminalCursor,
     origin: (usize, usize, usize),
-    copy: Option<Box<Cursor>>,
+    copy: Option<Box<CursorPosition>>,
 }
+
 impl Cursor {
-    pub fn new(x: usize, y: usize, width: usize) -> Self {
+    pub fn new(internal_cursor: TerminalCursor, x: usize, y: usize, width: usize) -> Self {
         Self {
-            screen_pos: (x, y),
-            buffer_pos: 0,
-            bounds: Bounds::new(y, (4, width)),
-            lock_pos: (4, y),
+            pos: CursorPosition {
+                screen_pos: (x, y),
+                buffer_pos: 0,
+                bounds: Bounds::new(y, (4, width)),
+                lock_pos: (4, y),
+            },
+            internal_cursor: internal_cursor,
             origin: (x, y, width),
             copy: None,
         }
     }
 
-    fn save_position(&mut self) {
-        self.copy = Some(Box::new(self.clone()));
+    pub fn save_position(&mut self) {
+        self.copy = Some(Box::new(self.pos.clone()));
     }
 
-    fn reset_position(&mut self) {
+    pub fn reset_position(&mut self) {
         if let Some(copy) = self.copy.take() {
-            *self = *copy.clone();
-            self.copy = Some(copy);
+            self.pos = *copy;
+            self.copy = Some(Box::new(self.pos.clone()));
         }
     }
 
-    fn move_right(&mut self) {
-        if self.screen_pos.0 == self.current_upper_bound() {
-            if self.bounds.contains(self.screen_pos.1 + 1) {
-                self.screen_pos.0 = self.bounds.lower_bound(self.screen_pos.1 + 1);
-                self.screen_pos.1 += 1;
+    // NOTE(smolck): Is `move_one_right` a good name for this?
```

Comment:
@sigmaSd That reason I have it like that is because of [this line here](https://github.com/smolck/IRust/blob/4986e7718d320616116770e326078ad3fc597386/src/irust.rs#L52), where we get the cursor from the newly created `Crossterm`. Is that an unnecessary extra step?

Reply:
Actually would you keep move_one_right I remembred Im planning to move buffer_pos to a Buffer struct so Cursor would only means screen cursor, same for move left, thx!